apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -73,54 +47,64 @@ const PageTemplateModal = withState( {
-				onRequestClose={ () => setState( { isOpen: false } ) }
-				className="page-template-modal"
-			>
-				<div className="page-template-modal__inner">
-					<form className="page-template-modal__form">
-						<fieldset className="page-template-modal__list">
-							<legend className="page-template-modal__intro">
-								<p>
-									{ __(
-										'Pick a Template that matches the purpose of your page.',
-										'full-site-editing'
-									) }
-								</p>
-								<p>
-									{ __(
-										'You can customize each Template to meet your needs.',
-										'full-site-editing'
-									) }
-								</p>
-							</legend>
-							<TemplateSelectorControl
-								label={ __( 'Template', 'full-site-editing' ) }
-								templates={ Object.values( verticalTemplates ).map( template => ( {
-									label: template.title,
-									value: template.slug,
-									preview: template.preview && template.preview.src,
-									previewAlt: template.preview && template.preview.alt,
-								} ) ) }
-								onClick={ newTemplate => {
-									setState( { isOpen: false } );
-									insertTemplate( verticalTemplates[ newTemplate ] );
-								} }
-							/>
-						</fieldset>
-					</form>
-				</div>
-			</Modal>
-		) }
-	</div>
-) );
+} )( ( { isOpen, verticalTemplates, setState } ) => {
+	const closeModal = () => {
+		setState( { isOpen: false } );
+		trackDismiss( vertical.id );
+	};
+	const selectTemplate = newTemplate => {
+		insertTemplate( verticalTemplates[ newTemplate ] );
+		setState( { isOpen: false } );
+		trackSelection( vertical.id, newTemplate );
+	};
+	return (
+		<div>
+			{ isOpen && (
+				<Modal
+					title={ __( 'Select Page Template', 'full-site-editing' ) }
+					onRequestClose={ closeModal }
+					className="page-template-modal"
+				>
+					<div className="page-template-modal__inner">
+						<form className="page-template-modal__form">
+							<fieldset className="page-template-modal__list">
+								<legend className="page-template-modal__intro">
+									<p>
+										{ __(
+											'Pick a Template that matches the purpose of your page.',
+											'full-site-editing'
+										) }
+									</p>
+									<p>
+										{ __(
+											'You can customize each Template to meet your needs.',
+											'full-site-editing'
+										) }
+									</p>
+								</legend>
+								<TemplateSelectorControl
+									label={ __( 'Template', 'full-site-editing' ) }
+									templates={ Object.values( verticalTemplates ).map( template => ( {
+										label: template.title,
+										value: template.slug,
+										preview: template.preview,
+										previewAlt: template.description,
+									} ) ) }
+									onClick={ newTemplate => selectTemplate( newTemplate ) }
+								/>
+							</fieldset>
+						</form>
+					</div>
+				</Modal>
+			) }
+		</div>
+	);
+} );
 
 registerPlugin( 'page-templates', {
 	render: function() {
 		return <PageTemplateModal />;
 	},
 } );
+
+trackView( vertical.id );
```

Comment:
Is this tracking the vertical being shown? It feels like this should be part of a component event cycle rather than on the page. I know at the moment we don't load the JS unless certain conditions met in the PHP, however, this could change and that would result in stats getting track just because the JS is loaded. 

Just a thought...

Reply:
I was wondering about that, too :)

Reply:
I was wondering if I can get away without rewriting the component fully… 🙈It's now refactored!

Reply:
Look at us having a full fledged component now! 🎉